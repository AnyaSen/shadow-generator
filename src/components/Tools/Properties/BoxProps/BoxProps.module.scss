@import "../../../../assets/scss/variables";
@import "../../../../assets/scss/breakpoints";

.BoxProps {
  margin: 0 $space-2;
  padding-top: $space-3;

  h1 {
    margin-bottom: $space-2;
  }

  @include lg {
    width: 45%;
  }

  @include md {
    width: 100%;
  }

  @include sm {
    padding-top: $space-2;
  }
}
