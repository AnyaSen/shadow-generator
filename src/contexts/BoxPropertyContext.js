import React, { useState, createContext } from "react";

export const BoxPropertyContext = createContext();

export const BoxPropertyProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(350);
  const [widthValue, setWidthValue] = useState(350);
  const [radiusValue, setRadiusValue] = useState(16);
  const [borderValue, setBorderValue] = useState(0);

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };

  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };

  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };

  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };

  return (
    <BoxPropertyContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue
      }}
    >
      {children}
    </BoxPropertyContext.Provider>
  );
};
