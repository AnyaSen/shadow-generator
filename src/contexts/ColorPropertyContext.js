import React, { useState, createContext } from "react";

export const ColorPropertyContext = createContext();

export const ColorPropertyProvider = ({ children }) => {
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ColorPropertyContext.Provider
      value={{
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ColorPropertyContext.Provider>
  );
};
