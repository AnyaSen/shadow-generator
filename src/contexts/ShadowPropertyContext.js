import React, { useState, createContext } from "react";

export const ShadowPropertyContext = createContext();

export const ShadowPropertyProvider = ({ children }) => {
  const [shiftHorizontically, setShiftHorizontically] = useState(20);
  const [shiftVertically, setShiftVertically] = useState(14);
  const [blurValue, setBlurValue] = useState(10);
  const [opacityValue, setOpacityValue] = useState(12);

  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };

  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };

  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };

  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };

  return (
    <ShadowPropertyContext.Provider
      value={{
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue
      }}
    >
      {children}
    </ShadowPropertyContext.Provider>
  );
};
