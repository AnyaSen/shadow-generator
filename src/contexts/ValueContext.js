import React, { useState, createContext } from "react";

export const ValueContext = createContext();

export const ValueProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(45);
  const [widthValue, setWidthValue] = useState(45);
  const [radiusValue, setRadiusValue] = useState(1);
  const [shiftHorizontically, setShiftHorizontically] = useState(1.25);
  const [shiftVertically, setShiftVertically] = useState(0.85);
  const [blurValue, setBlurValue] = useState(0.625);
  const [opacityValue, setOpacityValue] = useState(0.75);
  const [borderValue, setBorderValue] = useState(0);
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };
  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };
  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };
  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };
  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };
  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };
  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };
  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };
  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ValueContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue,
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue,
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ValueContext.Provider>
  );
};
