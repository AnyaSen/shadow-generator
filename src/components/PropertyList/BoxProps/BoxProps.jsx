import React, { useContext } from "react";
import Styles from "./BoxProps.module.scss";

import ValueSlider from "../../ValueSlider/ValueSlider";

import { BoxPropertyContext } from "../../../contexts/BoxPropertyContext";

export default function BoxProps() {
  const {
    heightValue,
    ChangeHeightValue,
    widthValue,
    ChangeWidthValue,
    borderValue,
    ChangeBorderValue,
    radiusValue,
    ChangeRadiusValue
  } = useContext(BoxPropertyContext);

  return (
    <div className={Styles.BoxProps}>
      <h1>BOX</h1>

      <ValueSlider
        heading="height"
        value={heightValue}
        handleChange={ChangeHeightValue}
        maxValue={350}
        minValue={16}
      />

      <ValueSlider
        heading="width"
        value={widthValue}
        handleChange={ChangeWidthValue}
        maxValue={350}
        minValue={16}
      />
      <ValueSlider
        heading="border"
        value={borderValue}
        handleChange={ChangeBorderValue}
        maxValue={12}
      />

      <ValueSlider
        heading="border-radius"
        value={radiusValue}
        handleChange={ChangeRadiusValue}
        maxValue={175}
      />
    </div>
  );
}
