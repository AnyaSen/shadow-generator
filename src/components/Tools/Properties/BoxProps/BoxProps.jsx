import React, { useContext } from "react";

import Styles from "./BoxProps.module.scss";
import { ValueContext } from "../../../../contexts/ValueContext";

import ValueSlider from "../../../ValueSlider/ValueSlider";

export default function BoxProps() {
  const {
    heightValue,
    ChangeHeightValue,
    widthValue,
    ChangeWidthValue,
    borderValue,
    ChangeBorderValue,
    radiusValue,
    ChangeRadiusValue
  } = useContext(ValueContext);

  return (
    <div className={Styles.BoxProps} id="box">
      <h1>BOX</h1>
      <ValueSlider
        heading="height"
        value={heightValue}
        handleChange={ChangeHeightValue}
        maxValue={60}
        minValue={1}
      />
      <ValueSlider
        heading="width"
        value={widthValue}
        handleChange={ChangeWidthValue}
        maxValue={70}
        minValue={1}
      />
      <ValueSlider
        heading="border"
        value={borderValue}
        handleChange={ChangeBorderValue}
        maxValue={3}
        minValue={0}
      />
      <ValueSlider
        heading="border-radius"
        value={radiusValue}
        handleChange={ChangeRadiusValue}
        maxValue={25}
      />
    </div>
  );
}
