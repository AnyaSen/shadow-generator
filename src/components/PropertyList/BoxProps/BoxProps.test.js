import React from "react";
import { render, cleanup } from "@testing-library/react";

import BoxProps from "./BoxProps";

import { BoxPropertyProvider } from "../../../contexts/BoxPropertyContext";

describe("<BoxProps />", () => {
  afterEach(cleanup);

  const createRenderTree = () => (
    <BoxPropertyProvider>
      <BoxProps />
    </BoxPropertyProvider>
  );

  test("Should return 5 children and correct text.", () => {
    const tree = createRenderTree();
    const text = "BOX";
    const { getByTestId, getByText, container } = render(tree);

    const shadowPropsContainer = getByTestId(`boxProps`);
    const shadowPropsHeading = getByText(text);

    expect(shadowPropsContainer.children.length).toEqual(5);
    expect(shadowPropsHeading).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
