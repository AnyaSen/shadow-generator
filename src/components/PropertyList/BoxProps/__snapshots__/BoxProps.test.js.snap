// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxProps /> Should return 5 children and correct text. 1`] = `
<div
  class="BoxProps"
  data-testid="boxProps"
>
  <h1>
    BOX
  </h1>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      height
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 55.08982035928144%;"
      />
      <input
        type="hidden"
        value="200"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="350"
        aria-valuemin="16"
        aria-valuenow="200"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 55.08982035928144%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      width
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 55.08982035928144%;"
      />
      <input
        type="hidden"
        value="200"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="350"
        aria-valuemin="16"
        aria-valuenow="200"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 55.08982035928144%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      border
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 0%;"
      />
      <input
        type="hidden"
        value="0"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="12"
        aria-valuemin="0"
        aria-valuenow="0"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </span>
  </div>
  <div
    class="slider"
    data-testid="Slider"
  >
    <p>
      border-radius
    </p>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary"
    >
      <span
        class="MuiSlider-rail"
      />
      <span
        class="MuiSlider-track"
        style="left: 0%; width: 9.142857142857142%;"
      />
      <input
        type="hidden"
        value="16"
      />
      <span
        aria-orientation="horizontal"
        aria-valuemax="175"
        aria-valuemin="0"
        aria-valuenow="16"
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
        data-index="0"
        role="slider"
        style="left: 9.142857142857142%;"
        tabindex="0"
      />
    </span>
  </div>
</div>
`;
